pkg/controller/backup_controller.go

```diff
@@ -951,7 +951,7 @@ func (c *backupController) deleteVolumeSnapshot(volumeSnapshots []*snapshotv1api
 			if vs.Status.BoundVolumeSnapshotContentName != nil &&
 				len(*vs.Status.BoundVolumeSnapshotContentName) > 0 {
 				vsc = vscMap[*vs.Status.BoundVolumeSnapshotContentName]
```

Comment:
Thanks for contribution.
This is useful prorection checking, but I think if vsc is nil, or cannot be found from the vscMap, we should record an error, then return.
In normal case, this shouldn't happen, but if this happens, it means VSC is deleleted accidently, and the external snapshot controller should recreate vsc sometime later, in this case, I think no need to delete vs, just leaving with error message should be enough.

Reply:
Have processed as what you suggest, please review again. Thanks!